keps/sig-network/3766-referencegrant/README.md

```diff
@@ -0,0 +1,492 @@
+	// Kind is the kind of the referent.
+	Kind string `json:"kind"`
+
+	// Name is the name of the referent. When unspecified, this policy
+	// refers to all resources of the specified Group and Kind in the local
+	// namespace.
+	//
+	// +optional
+	Name *string `json:"name,omitempty"`
+}
+```
+
+### Outstanding questions and clarifications
+
+This section lists some of the outstanding questions people have had about
+ReferenceGrant, and other items that we'll be clarifying in the godoc and other
+documentation as part of the transition to the new API group, keeping in mind
+that we will not be adding any new features, fields, or behavior, just clarifying
+what's done already.
+
+Also note that we don't consider any of these blockers for the general _idea_ of
+moving ReferenceGrant to the new API group, just notes to save discussion time.
+
+* Clarify that an implementation is required to reconcile ReferenceGrant for
+  specific `To` Kinds.
+* Corollary for future work, define how controllers interact. Is it a problem if
+  multiple controllers consume the same ReferenceGrant?
+* Status design is still pending, but it's currently expected that controllers
+  will indicate status on the _referring_ resources, not on ReferenceGrant itself.
+* Clarify that the expected operating model for implementations expects them to
+  have broad, read access to both ReferenceGrant and the specific `To` Kinds they
+  support, and then self-limit to only _use_ the relevant ones.
+* Decide whether to formally add `*` as a special value for Namespace, to mean
+  "all namespaces".
+
+### Test Plan
+
+[x] I/we understand the owners of the involved components may require updates to
+existing tests to make this code solid enough prior to committing the changes necessary
+to implement this enhancement.
+
+##### Prerequisite testing updates
+
+This is a net new resource to Kubernetes so it will not require any changes or
+additions to existing tests.
+
+##### Unit tests
+
+Unit tests will be used to cover:
+
+1. ReferenceGrant validation
+2. ReferenceGrant implementation library
+
+More details will be added as the details of the implementation library are
+clarified.
+
+##### Integration tests
+
+Because we are not bundling an implementation with this API, no integration
+tests are possible.
+
+##### e2e tests
+
+We will strongly encourage every API that uses ReferenceGrant to define
+conformance tests for their use of ReferenceGrant.
+
+### Graduation Criteria
+
+#### GA
+
+[x] Almost all of the fields and behavior have conformance test coverage.
+[x] Multiple conformant implementations.
+[x] Widespread implementation and usage.
+[ ] Conformance tests that exercise all ReferenceGrant API calls (not the actual implementation of the API).
+[ ] At least 6 months of soak time as a beta API (graduated to beta in Dec 2022).
+
+
+### Upgrade / Downgrade Strategy
+
+N/A
+
+### Version Skew Strategy
+
+Version skew is a bit different here since it won't be implemented in-tree.
```

Comment:
this bit probably needs updating now that this is proposing being in-tree, specifically around the library implementation and how it supports different versions / falls back if preferred versions are unavailable

Reply:
Tried to clarify this to make the distinction that the actual implementing controllers of this API will not be in-tree, even if they are importing a library from in-tree. As far as fall backs, do we have any standard guidelines here? This feels like it would be very similar to any other API.